enterprise/backend/test/metabase_enterprise/query_field_validation/api_test.clj

```diff
@@ -0,0 +1,166 @@
+(ns metabase-enterprise.query-field-validation.api-test
+  (:require
+   [clojure.string :as str]
+   [clojure.test :refer :all]
+   [mb.hawk.assert-exprs.approximately-equal :as hawk.approx]
+   [metabase.test :as mt]
+   [ring.util.codec :as codec]
+   [toucan2.tools.with-temp :as t2.with-temp]))
+
+(defn- do-with-test-setup [f]
+  (t2.with-temp/with-temp [:model/Table      {table  :id}  {:name "T"}
+                           :model/Card       {card-1 :id}  {:name "A"}
+                           :model/Card       {card-2 :id}  {:name "B"}
+                           :model/Card       {card-3 :id}  {:name "C"}
+                           :model/Card       {card-4 :id}  {:name "D"}
+                           :model/Field      {field-1 :id} {:active   false
+                                                            :name     "FA"
+                                                            :table_id table}
+                           :model/Field      {field-2 :id} {:active   false
+                                                            :name     "FB"
+                                                            :table_id table}
+                           :model/Field      {field-3 :id} {:active   false
+                                                            :name     "FC"
+                                                            :table_id table}
+                           ;; QFs not to include:
+                           ;; - Field is still active
+                           :model/QueryField {}            {:card_id  card-1
+                                                            :field_id (mt/id :orders :tax)}
+                           ;; - Implicit reference
+                           :model/QueryField {}            {:card_id            card-2
+                                                            :field_id           field-1
+                                                            :explicit_reference false}
+                           ;; QFs to include:
+                           :model/QueryField {qf-1 :id}    {:card_id  card-1
+                                                            :field_id field-1}
+                           :model/QueryField {qf-2 :id}    {:card_id  card-2
+                                                            :field_id field-2}
+                           :model/QueryField {qf-3 :id}    {:card_id  card-3
+                                                            :field_id field-3}]
+    (mt/with-premium-features #{:query-field-validation}
+      (f {:card-1 card-1
+          :card-2 card-2
+          :card-3 card-3
+          :card-4 card-4
+          :qf-1   qf-1
+          :qf-2   qf-2
+          :qf-3   qf-3}))))
```

Comment:
Macrology hat on, this seems like a common enough pattern. Should we make it a bit nicer to reuse?

How about two macros - firstly, for this return value:

```clojure
(anaphoric-provide f
  card-1 
  card-2
  ...)
```

And for using it in `with-test-setup`:

```clojure
(anaphoric-use-fn [card-1 card-2 ...] 
  ~@body)
```